### ISSUE TITLE
Add support for PEP 498 / fstrings

### ISSUE THREAD

**Issue by cmcginty** [google/yapf#383]
On my current homebrew install of yapf it chokes on parsing fstrings: 

```
$ echo "a = f'string'" | yapf
Traceback (most recent call last):
  File "/usr/local/bin/yapf", line 11, in <module>
    sys.exit(run_main())
  File "/usr/local/lib/python2.7/site-packages/yapf/__init__.py", line 296, in run_main
    sys.exit(main(sys.argv))
  File "/usr/local/lib/python2.7/site-packages/yapf/__init__.py", line 171, in main
    verify=args.verify)
  File "/usr/local/lib/python2.7/site-packages/yapf/yapflib/yapf_api.py", line 126, in FormatCode
    tree = pytree_utils.ParseCodeToTree(unformatted_source)
  File "/usr/local/lib/python2.7/site-packages/yapf/yapflib/pytree_utils.py", line 114, in ParseCodeToTree
    raise e
  File "<unknown>", line 1
    a = f'string'
                ^
SyntaxError: invalid syntax
```

**Comment by bastih** [comment]
The problem with fstrings is that they need to be fixed in 2to3 (fail in py 3.6 and trunk IIRC when I checked the last time) and as far as I can see, this cannot be tackled in yapf itself.

**Comment by ipfans** [comment]
This issue duplicated with https://github.com/google/yapf/issues/342
